### PR TITLE
ES-2279: [CLI] Upgrade CPI in network feature

### PR DESCRIPTION
**Pull request by anton-subbotin**
### [ES-2279](https://r3-cev.atlassian.net/browse/ES-2279)

[ES-2279]: https://r3-cev.atlassian.net/browse/ES-2279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by anton-subbotin** [comment]
build e2e

File: libs/corda-sdk/src/main/kotlin/net/corda/sdk/network/CpiUpgrade.kt [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    fun upgradeCpiOnVirtualNode(holdingId: ShortHash, cpiChecksum: Checksum) {
+        virtualNode.updateState(holdingId, VirtualNodeStateTransitions.MAINTENANCE)
+
+        waitUntilNoRunningFlows(holdingId)
+        upgradeCpiAndWaitForSuccess(holdingId, cpiChecksum)
+
+        virtualNode.updateState(holdingId, VirtualNodeStateTransitions.ACTIVE)
+    }
+
+    private fun waitUntilNoRunningFlows(holdingId: ShortHash, waitDuration: Duration = 30.seconds) {
```
Comment (tlawson3) [inline libs/corda-sdk/src/main/kotlin/net/corda/sdk/network/CpiUpgrade.kt:118]: Should this waitDuration be configurable in the fun upgradeCpiOnVirtualNode ? 
Currently I don't see how a user would override the 30 second default

File: tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/UpgradeCpi.kt [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+            val (cpiName, cpiVersion, groupId) = readAndValidateCpiMetadata()
+
+            val networkConfig = readNetworkConfig(networkConfigFile)
+
+            val targetVirtualNodes = getTargetVirtualNodes(networkConfig, cpiName, cpiVersion, groupId)
+
+            val cpiChecksum = uploadCpiAndGetChecksum()
+
+            val success = upgradeVirtualNodes(targetVirtualNodes, cpiChecksum)
```
Comment (tlawson3) [inline tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/UpgradeCpi.kt:98]: Pass the wait in here

Comment (tlawson3) [reply tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/UpgradeCpi.kt:98]: As the default is currently 10, you may need to do something like this: https://github.com/corda/corda-runtime-os/blob/14715e9231809a9df2e9dee8f4486b54533b2476/tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/BaseOnboard.kt#L106-L112

Or do you think the CLI should not allow the user to change the time in this context?

**Review by tlawson3 — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by nikinagy — APPROVED** [review]
LGTM

**Pull request merged**